VAMobile/src/screens/HomeScreen/ProfileScreen/PersonalInformationScreen/GenderIdentityScreen/GenderIdentityScreen.tsx

```diff
@@ -70,9 +78,19 @@ const GenderIdentityScreen: FC<GenderIdentityScreenProps> = ({ navigation }) =>
     setGenderIdentity(value)
   }
 
+  const updateGenderIdentity = () => {
+    if (genderIdentity) {
+      const mutateOptions = {
+        onSuccess: () => showSnackBar(snackbarMessages.successMsg, dispatch, undefined, true, false, true),
+        onError: () => showSnackBar(snackbarMessages.errorMsg, dispatch, updateGenderIdentity, false, true, true),
+      }
+      genderIdentityMutation.mutate(genderIdentity, mutateOptions)
+    }
+  }
+
```

Comment:
Instead of handling showing snack bars in the mutation hook, I moved this to the screen instead, so that UI related side effects can be isolated from the hook and API call. This feels cleaner since we don't have to reference translations and the snackbar messages in the mutation hook, and all UI logic can stay together